src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs

```diff
@@ -9036,54 +9036,21 @@ public static void Main()
             Assert.True(File.Exists(refDll));
 
             // The types and members that are included needs further refinement.
-            // ReferenceAssemblyAttribute is missing.
             // See issue https://github.com/dotnet/roslyn/issues/17612
             MetadataReaderUtils.VerifyPEMetadata(refDll,
-                new[] { "TypeDef:<Module>", "TypeDef:C" },
-                new[] { "MethodDef: Void Main()", "MethodDef: Void .ctor()" },
-                new[] { "CompilationRelaxationsAttribute", "RuntimeCompatibilityAttribute", "DebuggableAttribute" }
+                new[] { "TypeDefinition:<Module>", "TypeDefinition:C" },
+                new[] { "MethodDefinition:Void Main()", "MethodDefinition:Void .ctor()" },
+                new[] { "CompilationRelaxationsAttribute", "RuntimeCompatibilityAttribute", "DebuggableAttribute", "ReferenceAssemblyAttribute" }
                 );
 
-            VerifyNullReferenceException(refDir, "a.dll");
+            // Runtime won't load assembly with ReferenceAssemblyAttribute
+            ProcessUtilities.RunAndGetOutput(Path.Combine(refDir.Path, "a.dll"), startFolder: refDir.Path, expectedRetCode: -532462766);
```

Comment:
>ProcessUtilities.RunAndGetOutput(Path.Combine(refDir.Path, "a.dll"), startFolder: refDir.Path, expectedRetCode: -532462766); [](start = 12, length = 124)

At this point it is probably better to check the presence of the attribute and not attempt to run the code.